tor_core/helpers.py

```diff
@@ -0,0 +1,171 @@
+import logging
+import re
+import sys
+
+import prawcore
+import requests
+
+from tor_core import __version__
+from tor_core.strings import bot_footer
+
+
+def _(message):
+    """
+    Message formatter. Returns the message and the disclaimer for the
+    footer.
+
+    :param message: string. The message to be displayed.
+    :return: string. The original message plus the footer.
+    """
+    return bot_footer.format(message, version=__version__)
+
+
+def log_header(message):
+    logging.info('*' * 50)
+    logging.info(message)
+    logging.info('*' * 50)
+
+
+def clean_list(items):
+    """
+    Takes a list and removes entries that are only newlines.
+
+    :param items: List.
+    :return: List, sans newlines
+    """
+    cleaned = []
+    for item in items:
+        if item.strip() != '':
+            cleaned.append(item)
+
+    return cleaned
+
+
+def send_to_slack(message, config):
+    """
+    Sends a message to the ToR #general slack channel.
+
+    :param message: String; the message that is to be encoded.
+    :param config: the global config dict.
+    :return: None.
+    """
+    # if we have the api url loaded, then fire off the message.
+    # Otherwise, don't worry about it and just return.
+    if config.slack_api_url:
+        payload = {
+            'username': 'Kierra',
+            'icon_emoji': ':snoo:',
+            'text': message
+        }
+        requests.post(config.slack_api_url, json=payload)
+
+    return
+
+
+def explode_gracefully(error, config, bot_name=None):
+    """
+    A last-ditch effort to try to raise a few more flags as it goes down.
+    Only call in times of dire need.
+
+    :param bot_name: string; the name of the bot calling the method.
+    :param error: an exception object.
+    :param tor: the r/ToR helper object
+    :return: Nothing. Everything dies here.
+    """
+    logging.error(error)
+    if not bot_name:
+        bot_name = config.r.user.me().name
+    config.tor.message(
+        '{} BROKE - {}'.format(bot_name, error.__class__.__name__.upper()),
+        'Please check Bugsnag for the complete error.'
+    )
+    sys.exit(1)
+
+subreddit_regex = re.compile(
```

Comment:
Can we move this up to the top? Seems almost like a constant and it gets swallowed up when sandwiched between methods like this.